### PR TITLE
Async message + Fix (#580 #439 #1397 #699)

### PR DESCRIPTION
**Pull request by BenoitTallandier**
https://user-images.githubusercontent.com/22451567/130105959-ff038b8e-e5b1-4d5d-96af-aae0e209dfac.mov

https://user-images.githubusercontent.com/22451567/130105966-2a4bacae-2ea1-43de-993a-8e3e22af59cf.mov

### CODE REVIEW

**Comment by RomaricMourgues** [comment]
Can we keep you in the dev team ? :p
That’s great !

**Comment by chamerling** [comment]
This is great 🎉

File: twake/frontend/src/app/scenes/Apps/Messages/Message/Parts/MessageContent.tsx [hunk 1]
```diff
@@ -58,7 +59,9 @@ export default (props: Props) => {
   const deleted = props.message.subtype === 'deleted';
 
   const showEdition = !props.linkToThread && props.edited;
-
+  const messageIsLoading = (props.message as any)._creating || (props.message as any)._updating;
+  const messageSaveFailed =
+    (props.message as any)._failed || props.message.id == '6dec25e0-fc2e-11eb-a9c3-01401e397361';
```
Comment (chamerling) [inline twake/frontend/src/app/scenes/Apps/Messages/Message/Parts/MessageContent.tsx:64]: what is it?

File: twake/frontend/src/app/scenes/Apps/Messages/Message/Parts/MessageHeader.tsx [hunk 2]
```diff
@@ -161,6 +163,16 @@ export default class MessageHeader extends Component<Props, State> {
...
-  }
-}
+      {props.loading && (
+        <div className="loading">
+          <Loader color="#999" className="message_header_loader" />
+        </div>
+      )}
+      {props.failed && !props.loading && (
+        <div className="alert_failed_icon">
+          <AlertTriangle size={16} />
```
Comment (chamerling) [inline twake/frontend/src/app/scenes/Apps/Messages/Message/Parts/MessageHeader.tsx:173]: What happens then? DO we have a retry or edit button?

File: twake/frontend/src/app/services/Depreciated/Collections/Collection.js [hunk 3]
```diff
@@ -675,13 +675,23 @@ export default class Collection extends Observable {
...
 
               this.share(res.data.object);
               if (callback) callback(res.data.object);
             }
           }
 
           this.object_buffer_flush(res.data.object);
+        } else {
+          if (callback) {
+            console.log('gga !!error!!');
```
Comment (chamerling) [inline twake/frontend/src/app/services/Depreciated/Collections/Collection.js:688]: ggaaaaaa!!!!!

File: twake/frontend/src/app/scenes/Apps/Messages/Message/Parts/MessageContent.tsx (hunk 1 above)
Comment (BenoitTallandier) [reply twake/frontend/src/app/scenes/Apps/Messages/Message/Parts/MessageContent.tsx]: My bad, it's fixed ^^

**Review by RomaricMourgues — CHANGES_REQUESTED** [review]
You must solve the big changes on the languages files or it will break Weblate :)

**Comment by BenoitTallandier** [comment]
> You must solve the big changes on the languages files or it will break Weblate :)

Are you sure ? I checked every locale file has 2 space as indentation. Only en and fr has 4. Do we want to keep this difference ? :)

**Comment by RomaricMourgues** [comment]
It seems weblate is doing this itself: https://github.com/linagora/Twake/pull/1448/files
I prefer to reduce the number of changes we do to the language file to avoid merge issues with weblate if possible.

**Comment by RomaricMourgues** [comment]
@BenoitTallandier it was a good idea, thanks for trying to fix this indent issue :p

**Pull request merged**